Source/Core/HeadingPitchRoll.js

```diff
@@ -0,0 +1,96 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './freezeObject',
+    './Math',
+    './defineProperties'
+], function (
+        defaultValue,
+        defined,
+        DeveloperError,
+        freezeObject,
+        CesiumMath,
+        defineProperties) {
+    "use strict";
+
+    /**
+     * A rotation expressed as a heading, pitch, and roll. Heading is the rotation about the
+     * negative z axis. Pitch is the rotation about the negative y axis. Roll is the rotation about
+     * the positive x axis.
+     * @alias HeadingPitchRoll
+     * @constructor
+     *
+     * @param {Number} [heading=0.0] The heading component in radians.
+     * @param {Number} [pitch=0.0] The pitch component in radians.
+     * @param {Number} [roll=0.0] The roll component in radians.
+     */
+    var HeadingPitchRoll = function (heading, pitch, roll) {
+        this.heading = defaultValue(heading, 0.0);
+        this.pitch = defaultValue(pitch, 0.0);
+        this.roll = defaultValue(roll, 0.0);
+    };
+
+    /**
+     * Computes the heading, pitch and roll from a quaternion (see http://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles )
+     *
+     * @param {Quaternion} quaternion The quaternion from which to retrieve heading, pitch, and roll, all expressed in radians.
+     * @param {Quaternion} [result] The object in which to store the result. If not provided, a new instance is created and returned.
+     * @returns {HeadingPitchRoll} The modified result parameter or a new HeadingPitchRoll instance if one was not provided.
+     */
+    HeadingPitchRoll.fromQuaternion = function (quaternion, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(quaternion)) {
+            throw new DeveloperError('quaternion is required');
+        }
+        //>>includeEnd('debug');
+        result = defaultValue(result, new HeadingPitchRoll());
```

Comment:
Use the following, otherwise a `HeadingPitchRoll` instance is allocated no matter what:
```javascript
if (!defined(result)) {
    result = new HeadingPitchRoll();
}
```